js/multilingual_survey_settings.js

```diff
@@ -460,6 +512,33 @@ Multilingual.saveSurveySettings = function() {
 		})
 	})
 	
+	// remember hide option setting
+	if ($("#hide_lang").is(':checked')) {
+		data.collections.basic_settings.hide_lang = '1';
+	} else {
+		data.collections.basic_settings.hide_lang = null;
+	}
+	
+	// remove previously cached settings for this language
+	Multilingual.settings_to_save = Multilingual.settings_to_save.filter(lang_settings => lang_settings.language != language)
+	
+	// append new survey settings for this language
+	Multilingual.settings_to_save.push(data)
+}
+
+Multilingual.saveSurveySettings = function() {
```

Comment:
I assume this function has been pretty well tested manually.

Reply:
Yes.